presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -429,7 +432,7 @@ public ReadContext init(
             ImmutableList.Builder<parquet.schema.Type> fields = ImmutableList.builder();
             for (HiveColumnHandle column : columns) {
                 if (!column.isPartitionKey() && column.getHiveColumnIndex() < messageType.getFieldCount()) {
-                    fields.add(messageType.getType(column.getName()));
+                    fields.add(getParquetType(configuration, column, messageType));
```

Comment:
Yes. That is what I was thinking.  When I have column name access on, reordering should work.  When I have column index access on, metastore column renames work (but no reorder, just like normal hive).

So are going to update this PR so that column name access works with reordering?